src/lib/Form/Type/Content/Location/ContentLocationSetMainType.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace EzSystems\EzPlatformAdminUi\Form\Type\Content\Location;
+
+use EzSystems\EzPlatformAdminUi\Form\Data\Content\ContentUpdateMainLocationData;
+use EzSystems\EzPlatformAdminUi\Form\Type\Content\ContentInfoType;
+use EzSystems\EzPlatformAdminUi\Form\Type\Content\LocationType;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\Extension\Core\Type\SubmitType;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use eZ\Publish\API\Repository\LocationService;
+use eZ\Publish\API\Repository\Values\Content\Location;
+
+class ContentLocationSetMainType extends AbstractType
```

Comment:
We do not have any formal guidelines for Type/Data class naming but to keep things streamlined:

* Corresponding data class is named `ContentUpdateMainLocationData` or as proposed by me: `ContentMainLocationUpdateData`. The type should be named after the data class: `ContentMainLocationUpdateType`.
* With above change you should also change submit button name as it should be named after action name (which now is `set` but should be `update`).